proparse/src/main/java/org/prorefactor/treeparser/ParseUnit.java

```diff
@@ -133,6 +137,24 @@ public TokenStream lex() throws IOException {
     return lexer.getANTLR2TokenStream(false);
   }
 
+  public void preprocess() throws IOException {
+    ProgressLexer lexer = new ProgressLexer(session, file.getPath(), fileNameList, false);
+    TokenStream stream = lexer.getANTLR2TokenStream(false);
+    try (FileWriter writer = new FileWriter("foo.preprocess")){
+      ProToken tok = (ProToken) stream.nextToken();
+      while (tok.getType() != Token.EOF_TYPE) {
+        if ((tok.getNodeType() == ABLNodeType.AMPSCOPEDDEFINE) || (tok.getNodeType() == ABLNodeType.AMPGLOBALDEFINE)) {
+//            writer.write("prepro");
+        } else {
+          writer.write(tok.getText());
+        }
+        tok = (ProToken) stream.nextToken();
+      }
+    } catch (TokenStreamException uncaught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00108)